### PR TITLE
web-wallet: Update Buttons to match DS

### LINKED ISSUE #1606: web-wallet: Refactor Buttons CSS and fix design
The icon buttons design has been updated:

Screenshot 2024-03-27 at 17 52 47

The hover state has background color of `--secondary-color` and main colour `--on-secondary-color`.

Moreover, we have reduced the number of variants to three from four, which presents an opportunity to streamline our code by eliminating unused elements.

Additionally, the CSS for the Button could benefit from a revision for better readability and comprehension.

### PR DESCRIPTION
**Pull request by nortonandreev**
Resolves #1606

Screenshot 2024-04-04 at 10 01 21
Screenshot 2024-04-04 at 10 00 53

### CODE REVIEW

**Comment by nortonandreev** [comment]
https://github.com/dusk-network/rusk/assets/34094428/0d93fa45-4c34-4954-9711-811d525ccc6c

Not sure why the animation for these two variants look different on dark mode. 🤔

File: web-wallet/src/style/dusk/language.css [hunk 1]
```diff
@@ -53,26 +53,30 @@
   --control-small-font-size: 0.875em;
   --control-small-padding: 0.625em 0.875em;
 
-  --control-bg-color: #fff;
-  --control-bg-color-hover: var(--silver-metallic);
+  --non-button-control-bg-color: #fff;
+  --non-button-control-text-color: var(--smokey-black);
```
Comment (ascartabelli) [inline web-wallet/src/style/dusk/language.css:57]: I don't think it's worth it to add complexity to this `--control` variables by adding a `--non-button` case.

In our current design document even the selects have a smokey black background in light mode: the only "controls" keeping a white background are textboxes for now, if I'm not mistaken.

I think we should let @laremas think about it and then have a chat with him, but my guts tell me that we should probably get rid of these `--control` vars altogether.

Having said that: no need to change here in this PR, as this can be a temporary step while we formalize things.

An extra issue to keep track of the problem is needed anyway, though.

(cc @kieranhall @deuch13 )

File: web-wallet/src/style/dusk/language.css [hunk 2]
```diff
@@ -120,12 +124,15 @@
...
-  --control-secondary-bg-color: var(--light-grey);
-  --control-secondary-text-color: var(--smokey-black);
+  --non-button-control-bg-color: var(--smokey-black);
+  --non-button-control-text-color: var(--light-grey);
+
+  --control-bg-color: var(--light-grey);
+  --control-text-color: var(--smokey-black);
+  --control-secondary-bg-color: var(--smokey-black);
+  --control-secondary-bg-color-alt: var(--black);
+  --control-secondary-border-color-alt: var(--black);
```
Comment (ascartabelli) [inline web-wallet/src/style/dusk/language.css:134]: See other comments about "alt" and `--black`.

**Review by ascartabelli — CHANGES_REQUESTED** [review]

File: web-wallet/src/style/dusk-components/button.css [hunk 3]
```diff
@@ -12,16 +13,67 @@
...
+  transition:
+    0.2s ease-in-out background-color,
+    border-color,
+    color,
+    transform;
 }
 
 .dusk-button:not(.dusk-anchor-button) {
   cursor: default;
 }
```
[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/style/dusk-components/button.css:25]: Yeah, I don't it's needed!

File: web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte [hunk 4]
```diff
@@ -24,17 +24,21 @@
...
   /** @type {String | Undefined} */
   export let text = undefined;
 
+  /** @type {Boolean} */
+  export let alt = false;
+
   $: classes = makeClassName([
     "dusk-anchor-button",
-    `dusk-anchor-button--variant--${variant}`,
-    `dusk-anchor-button--size--${size}`,
```
[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte:35]: As we discussed last week, `--{mod name}--{mod-value}` is an alternative naming, not the BEM in its core form.

As I have said, I don't have that strong preference, but I would stick to BEM as it is, as it's clearer.

cc: @kieranhall @deuch13

File: web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte [hunk 5]
```diff
@@ -24,17 +24,21 @@
...
 
   /** @type {ButtonVariant} */
-  export let variant = "secondary";
+  export let variant = "primary";
 
   /** @type {String | Undefined} */
   export let text = undefined;
 
+  /** @type {Boolean} */
+  export let alt = false;
```
[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte:33]: Yes, right now, the only exception is the Settings button on the Dashboard layout.

I think these points are valid, we can refactor this so that the alt variant is only applied to this instance of the button.

File: web-wallet/src/routes/components-showcase/Buttons.svelte [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    disabled={true}
+    icon={{ path: mdiHome }}
+    size="small"
+    text="Tertiary - Disabled"
+    variant="tertiary"
+  />
+</section>
+<br />
+<section>
+  <Button icon={{ path: mdiHome, size: "large" }} text="Primary - Large" />
```
[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/routes/components-showcase/Buttons.svelte:132]: I checked, doesn't seem like it's used anywhere. Have removed it! Thanks for pointing this out!

File: web-wallet/src/style/dusk/colors.css [hunk 7]
```diff
@@ -4,6 +4,7 @@
   --granite-grey: #636167;
   --jet: #2e2d30;
   --smokey-black: #101010;
+  --black: #000000;
```
[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/style/dusk/colors.css:7]: Changed.

File: web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte (hunk 4 above)
[the remark this answers is not in the document]
Comment (ascartabelli) [reply web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte:35]: > As we discussed last week, `--{mod name}--{mod-value}` is an alternative naming, not the BEM in its core form.
> 
> As I have said, I don't have that strong preference, but I would stick to BEM as it is, as it's clearer.

Actually no, in our discussion @deuch13 and me pointed out that "classic" BEM is `block-name__elem-name_mod-name_mod-val`, while the alternative syntax we chose to use is `block-name__elem-name--mod-name--mod-val`.
You can omit the `mod_val` bit for "booleans" with a "true" value.

See the [naming conventions](https://en.bem.info/methodology/naming-convention/).

File: web-wallet/src/style/dusk-components/button.css [hunk 8]
```diff
@@ -38,130 +90,17 @@
   text-overflow: ellipsis;
 }
 
-.dusk-button:enabled:active,
-.dusk-anchor-button:not(.dusk-anchor-button--disabled):active {
-  transform: scale(0.98, 0.98);
```
[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/style/dusk-components/button.css:43]: Waiting on @laremas to confirm, my understanding is that hover and active states are the same.

File: web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte (hunk 4 above)
[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte]: Have reverted the change. Still not sure what we gain by having the extra complexity.

I have been following [this](https://getbem.com) guide.

File: web-wallet/src/style/dusk-components/button.css (hunk 8 above)
[the remark this answers is not in the document]
Comment (laremas) [reply web-wallet/src/style/dusk-components/button.css:43]: mmm, not sure i get your question. [Here](https://www.figma.com/file/mkLSb0BV2jJa5blFj8VxUJ/Dusk-2.1-(variables)?type=design&node-id=192%3A1120&mode=design&t=QQ1Ga8zLizzdqmYW-1) are the buttons states. Hover state is different from Active.
Screenshot 2024-04-09 at 12 16 29

[the remark this answers is not in the document]
Comment (ascartabelli) [reply web-wallet/src/style/dusk-components/button.css:43]: > mmm, not sure i get your question. [Here](https://www.figma.com/file/mkLSb0BV2jJa5blFj8VxUJ/Dusk-2.1-(variables)?type=design&node-id=192%3A1120&mode=design&t=QQ1Ga8zLizzdqmYW-1) are the buttons states. Hover state is different from Active. 

Maybe you're referring to the "focused" state?

By "active" we mean "pressed": think of a toggle button where the button stays pressed.

@nortonandreev we need the active state for that one too in the styles.

Also, as we're correcting our BEM classes, this

```js
type === "toggle" && active ? "dusk-button--type--toggle--active" : ""
```

should probably become:

```js
type === "toggle" && active ? "dusk-button--active" : ""
```

[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/style/dusk-components/button.css:43]: > Maybe you're referring to the "focused" state?

It looks like it, also the border to the state was just added – I don't think it was there before.

Still not getting what the difference between the active and the focus states should be, if any.

File: web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte (hunk 4 above)
[the remark this answers is not in the document]
Comment (kieranhall) [reply web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte]: I think it's just important that we follow one standard.  I think following the [naming conventions ]([naming conventions](https://en.bem.info/methodology/naming-convention/))is the right way to go.

File: web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte (hunk 5 above)
[the remark this answers is not in the document]
Comment (kieranhall) [reply web-wallet/src/lib/dusk/components/AnchorButton/AnchorButton.svelte]: I think removing the `alt` prop and relying on contextual styling is the way to go.

File: web-wallet/src/style/dusk-components/button.css (hunk 8 above)
[the remark this answers is not in the document]
Comment (kieranhall) [reply web-wallet/src/style/dusk-components/button.css:43]: > Still not getting what the difference between the active and the focus states should be, if any.

I think @laremas should clarify on this.

**Review by kieranhall — COMMENTED** [review]
Some topics in the comments to be addressed but overall looks good!

File: web-wallet/src/style/dusk/colors.css (hunk 7 above)
[the remark this answers is not in the document]
Comment (kieranhall) [reply web-wallet/src/style/dusk/colors.css]: The Figma for the design variables does include black and white.  These may come in hand when documenting the colour palette when using a tool like Histoire, so I am not against keeping them in, despite agreeing with you that they are not _really_ needed.

File: web-wallet/src/style/dusk-components/button.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (kieranhall) [reply web-wallet/src/style/dusk-components/button.css]: If this just reinforces the default browser behaviour, it can be removed.

File: web-wallet/src/style/dusk-components/button.css (hunk 8 above)
[the remark this answers is not in the document]
Comment (kieranhall) [reply web-wallet/src/style/dusk-components/button.css:43]: I've asked @laremas to chime in.

File: web-wallet/src/style/dusk-components/button.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (ascartabelli) [reply web-wallet/src/style/dusk-components/button.css]: > If this just reinforces the default browser behaviour, it can be removed.

Absolutely yes: I think I added last summer to override tailwind in a project and then forgot about it.

File: web-wallet/src/style/dusk-components/button.css (hunk 8 above)
[the remark this answers is not in the document]
Comment (laremas) [reply web-wallet/src/style/dusk-components/button.css:43]: The Active (button remains pressed) state has 1px solid cornflower blue border appearing and stays there when user is interacting with it. No transformation is needed now (meananing no animation will be rendered).

[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/style/dusk-components/button.css:43]: It's not going to make a difference, as the hover state is already using the cornflower, so having the same border won't do much.

Screenshot 2024-04-10 at 12 31 50

The question I am trying to get an answer to is do we need an active state in the first place. I'm not sure it's needed...

[the remark this answers is not in the document]
Comment (ascartabelli) [reply web-wallet/src/style/dusk-components/button.css:43]: I already answered to you yesterday about it, I'll paste my answer here again though:

> I'll make an example
think of a settings button
in some cases you want the button to open a panel (drawer) while staying in the same page
in this case a switch isn't the correct fit
and here is where the button type="toggle" comes in
but it needs an active state that indicates, visually, that the button is in its pressed state
another common use case is a mute button in a video player
(in this case the active state is "special" because you often just change icon)
in a nutshell I think it's nice to have it ready

[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/style/dusk-components/button.css:43]: Yes, but we don't have this use case. I removed the code related to the toggle.

And even if we did, why would this have a different appearance?

[the remark this answers is not in the document]
Comment (ascartabelli) [reply web-wallet/src/style/dusk-components/button.css:43]: > Yes, but we don't have this use case. I removed the code related to the toggle.

We don't have the use case *now*. The button is meant to be a generic component.
And, please, before removing wait for the others, especially @laremas , to barge in.

> And even if we did, why would an active/toggled button have a different appearance from the hover state?

Because they have a different meaning, so it's not always appropriate. But this, again, it's @laremas decision.

[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/style/dusk-components/button.css:43]: > > Yes, but we don't have this use case. I removed the code related to the toggle.
> 
> We don't have the use case _now_. The button is meant to be a generic component. And, please, before removing wait for the others, especially @laremas , to barge in.
> 

My understanding is that this wasn't designed in the first place, so not sure why it was in the code to start off with.

[the remark this answers is not in the document]
Comment (laremas) [reply web-wallet/src/style/dusk-components/button.css:43]: The all concern come from the fact that we wanted to avoid the use of a shading when the component is used, as it was a placeholder. We decided to do so cause it was the only place where a shading appeared in the design. the changes were just to change the shading to a 1px ( 0.0625em) cornflower blue solid border.

[the remark this answers is not in the document]
Comment (laremas) [reply web-wallet/src/style/dusk-components/button.css:43]: I would also close this and open dedicated issues for the changes required, if anyit is hard to folllow the differents threads on this one. @Fulviuus

[the remark this answers is not in the document]
Comment (ascartabelli) [reply web-wallet/src/style/dusk-components/button.css:43]: > The all concern come from the fact that we wanted to avoid the use of a shading when the component is used, as it was a placeholder. We decided to do so cause it was the only place where a shading appeared in the design. the changes were just to change the shading to a 1px ( 0.0625em) cornflower blue solid border.

@nortonandreev wasn't referring to what was the focus state, but rather on the opportunity of having an active one.
I do think is needed (rationale above) even if there is no occurence of a "toggle button" in the wallet or explorer right now.
Also I don't see the need of removing the functionality which is already present and just needs a new style if we remove the transform.

[the remark this answers is not in the document]
Comment (laremas) [reply web-wallet/src/style/dusk-components/button.css:43]: Ok i realized my comment was not related about the questions made. 

So i would put back the states as they was to cover all the possibilities. Also transformations as they were coded.

@nortonandreev We can def refine it later adding, changing or removing stuff. 

Generally i would avoid removing things here and there also if they appear to be useless at first glance, design grown besides being very minimal, all things live a balance where changing a little one will impact other as a tsunami.

[the remark this answers is not in the document]
Comment (nortonandreev) [reply web-wallet/src/style/dusk-components/button.css:43]: I am not willing to spend more time on this issue – progress with it as you wish.

I have done my best to refactor the code so that it's easy to follow and maintain, while using the design system in place as a source of truth on how the component should appear and behave.

> Also I don't see the need of removing the functionality which is already present

This is not a valid argument, as the functionality present was added as a matter of a personal preference and an imaginary use case, rather than what is actually needed.

> just needs a new style

If it was as simple as that, it wouldn't have taken so long to come up with a "new style". We are creating an issue that didn't exist in the first place.

---

I would also argue that a Button and Toggle are two different components with different purpose. To start off with, having the `active` prop within the button is confusing, as a button can be disabled, so the `active` attribute is ambiguous in this context. Second, code, such as the one below, which was present in the previous revision of the Button, screams an anti-pattern to me:

Screenshot 2024-04-10 at 17 54 58

If there is a code like this, perhaps something is wrong logically in the first place.



cc: @Fulviuus @kieranhall

[the remark this answers is not in the document]
Comment (kieranhall) [reply web-wallet/src/style/dusk-components/button.css:43]: Let's do as @laremas suggests and put back the removed styles and move past this. As a follow up, @nortonandreev will propose a better solution in Duskit using multiple components that we can adopt after the design language foundations are established.

**Review by kieranhall — CHANGES_REQUESTED** [review]
Made a request to resolve the removed styles discussion.

**Comment by nortonandreev** [comment]
I have looked at the behavior that we currently have on production/main.

Screenshot 2024-04-11 at 17 45 38

If I use the components-showcase and change one of the Buttons to type toggle, I get a scaled down appearance of the button (with the `active` attribute being set to `true`).

If I change the Button to be only icon, without a label, I am seeing different behavior – the button is both scaled down, and uses the cornflower background (which comes from the fact that the button is icon-only, but I find this unpredictable – buttons of the same variant should look the same).

Screenshot 2024-04-11 at 17 48 20

Additionally, to toggle the active state, one has to add an on:click listener to the Button and toggle the active state from within – which is a bit awkward for a component we pretend to be a Toggle – if it was, the prop should have been updated on click automatically, which obviously we can't do if it's also a Button which has a different purpose.

https://github.com/dusk-network/rusk/assets/34094428/6307f954-57cc-41c9-8cf6-d1544f563b82

---

Furthermore, what we had for the icon buttons (of some variants) until now, is the following:

https://github.com/dusk-network/rusk/assets/34094428/a23cee5f-bc01-463d-8b0a-4833277a5632

, thus the scale transition.

In the latest iteration of the design system, the icon buttons have been unified so that they have the same appearance as the labeled buttons of the same variant (preview below):

https://github.com/dusk-network/rusk/assets/34094428/6b39d04c-0785-419c-b732-34c5d57fb343

Here are the labeled buttons:

https://github.com/dusk-network/rusk/assets/34094428/49873d53-7167-44ae-a411-3ab0616aaf02

and the transition from the icon buttons have been removed, so to match the look of the labeled buttons.

---

What I am trying to say is that we currently (on production), have way too many variants of the buttons which are often unpredictable, at least to me. This most probably means it will be the case for other developers too.

I think we should avoid bringing the scale transitions back so to have the same look and feel across the buttons, no matter if labeled or icon-only.

This matches [Figma](https://www.figma.com/file/mkLSb0BV2jJa5blFj8VxUJ/Dusk-2.1-(variables)?type=design&node-id=488%3A284&mode=design&t=MSvOu47xcmblxgaE-1).

Screenshot 2024-04-11 at 18 18 21

It can be seen that an `idle` and a `hovered` states have the same size (true both for labeled and icon-only buttons).

---

When it comes to the active button of a type toggle, as a middle ground I suggest I bring back the functionality for the time being (so that the button can be used as a toggle, if needed, and we separate this in Duskit later on), with the active state being the same as the hover state of the button.

https://github.com/dusk-network/rusk/assets/34094428/cba7c661-af24-4fe0-bee8-089aa3f283d2

Perhaps, in this case, it is also fine to scale down the button a bit, to convey the "pressed" state, but this depends on how these toggle would potentially be used. As we don't have such use case now, I cannot say which solution is better.

https://github.com/dusk-network/rusk/assets/34094428/c56fed3f-4d1f-428d-8626-f680c37c77c4

File: web-wallet/src/style/dusk-components/button.css (hunk 8 above)
[the remark this answers is not in the document]
Comment (kieranhall) [reply web-wallet/src/style/dusk-components/button.css:43]: > I would also close this and open dedicated issues for the changes required, if any. It is hard to folllow the different threads on this one. @Fulviuus

I've opened an [issue](https://github.com/dusk-network/duskit/issues/66) on Duskit.

**Comment by nortonandreev** [comment]
I have added back the toggle option of the Button with the scale transition.

Have a look now.

**Review by kieranhall — APPROVED** [review]
A new issue will be opened to address the icon button scaling issue.

**Review by ascartabelli — APPROVED** [review]
